R/cluster.R

```diff
@@ -387,31 +372,113 @@ getCluster <- function(clusterName) {
 
   nodes <- rAzureBatch::listPoolNodes(clusterName)
 
-  currentNodeCount <- 0
   if (!is.null(nodes$value) && length(nodes$value) > 0) {
-    nodesStatus <- .processNodeCount(nodes)
-
-    currentNodeCount <- nodesStatus$currentNodeCount
-    nodesWithFailures <- nodesStatus$nodesWithFailures
+    nodesInfo <- .processNodeCount(nodes)
+    nodesState <- nodesInfo$nodesState
+    nodesWithFailures <- nodesInfo$nodesWithFailures
+
+    if (verbose == TRUE) {
+      cat("\nnodes:", fill = TRUE)
+      cat(sprintf("\tidle:                %s", nodesState$idle), fill = TRUE)
+      cat(sprintf("\tcreating:            %s", nodesState$creating), fill = TRUE)
+      cat(sprintf("\tstarting:            %s", nodesState$starting), fill = TRUE)
+      cat(sprintf("\twaitingforstarttask: %s", nodesState$waitingforstarttask), fill = TRUE)
+      cat(sprintf("\tstarttaskfailed:     %s", nodesState$starttaskfailed), fill = TRUE)
+      cat(sprintf("\tpreempted:           %s", nodesState$preempted), fill = TRUE)
+      cat(sprintf("\trunning:             %s", nodesState$running), fill = TRUE)
+    }
 
     .showNodesFailure(nodesWithFailures)
   }
 
-  if (currentNodeCount >= totalNodes) {
-    config <- getOption("az_config")
-    cat("Your cluster has been registered.", fill = TRUE)
-    cat(sprintf("Dedicated Node Count: %i", pool$targetDedicatedNodes),
-        fill = TRUE)
-    cat(sprintf("Low Priority Node Count: %i", pool$targetLowPriorityNodes),
-        fill = TRUE)
+  cat("Your cluster has been registered.", fill = TRUE)
 
-    config$poolId <- clusterName
-    options("az_config" = config)
-    return(getOption("az_config"))
-  } else {
-    cat("Your cluster is not ready yet.", fill = TRUE)
-    return (NULL)
+  config <- getOption("az_config")
+  config$targetDedicatedNodes <- pool$targetDedicatedNodes
+  config$targetLowPriorityNodes <- pool$targetLowPriorityNodes
+  cat(sprintf("Dedicated Node Count: %i", pool$targetDedicatedNodes),
+      fill = TRUE)
+  cat(sprintf("Low Priority Node Count: %i", pool$targetLowPriorityNodes),
+      fill = TRUE)
+
+  config$poolId <- clusterName
+  options("az_config" = config)
+  return (config)
+}
+
+#' Get a list of clusters by state from the given filter
+#'
+#' @param filter A filter containing cluster state
+#'
+#' @examples
+#' \dontrun{
+#' getClusterList()
+#' }
+#' @export
+getClusterList <- function(filter = NULL) {
+  filterClause <- ""
+
+  if (!is.null(filter)) {
+    if (!is.null(filter$state)) {
+      for (i in 1:length(filter$state)) {
```

Comment:
is 'filter' a list or a vector? If so, can't se just do 
```R
filterClause <- paste(filter, " or ")
```

instead of this long for loop?

Reply:
that won't work, filter is a list of vectors, for example:
  filter <- filter <- list()
  filter$state <- c("active", "deleting")
  getClusterList(filter)
paste(filter, " or ") will produce "c('active', 'deleting') or " which is not a valid clause